java-type-checker/java_type_checker/types.py

```diff
@@ -133,25 +140,49 @@ def method_named(self, name):
                     pass
             raise NoSuchJavaMethod("{0} has no method named {1}".format(self.name, name))
 
+    def is_subtype_of(self, other):
+        if other == self:
+            return True
+        for supertype in self.direct_supertypes:
+            if supertype.is_subtype_of(other):
+                return True
```

Comment:
You did this on one line. I think mine is a bit more clear, but they work the same, and it could totally be argued that yours is better. However, the any() function you used is really useful.